NMS-14635: Add alt text to Operation (pt. 4) (#5743)
OpenNMS/opennms: docs/modules/operation/pages/deep-dive/admin/webui/jmx-config-generator/cli.adoc

| false
|===

=== generate-conf

The `generate-conf` sub-command can be used to generate a valid `jmx-datacollection-config.xml` file for a given set of MBean objects queried from an MBean server.
The following example generates a `myconfig.xml` file for MBean objects in the `java.lang` domain of `myserver`, on port 7199.
It uses the `myusername`/`mypassword` credentials for access:

[source, console]
----
jmx-config-generator generate-conf --host myserver --username myusername --password mypassword --port 7199 "java.lang:*" --output myconfig.xml
Dictionary entries loaded: '18'
----

NOTE: You must define either a URL or a hostname and port to connect to a JMX server.

The following command line options are available for the `generate-conf` sub-command:

[cols="1,3,1"]
|===
| Option/Argument	| Description	| Default Value

| <attribute id>
| A list of attribute IDs to be included in the new configuration file.
| _Empty_

| --dictionary <file>
| Path to a dictionary file containing replacements for attribute names and parts of MBean attributes.

> For each line in the file, a replacement should be defined (for example, `Auxillary:Auxil`).

Author: ```suggestion
Each line in the file must include a replacement definition (for example, `Auxillary:Auxil`).
```

<output omitted>
----

The following command line options are available for the `generate-graph` sub-command:

[cols="2,4,1"]
|===
| Option/Argument	| Description	| Default Value

| --input <jmx-datacollection.xml>

> | Configuration file to be used as the input when generating the graph properties file.

Reviewer: ```suggestion
| Configuration file to use as the input when generating the graph properties file.
```

| _Empty_

| --output <file>
| Output file name for the generated graph properties file.
| _Empty_

| --print-template
| Prints the default template.
| false

| --template <file>

> | Name of the template file to be used to generate the graph properties.

Reviewer: ```suggestion
| Name of the template file to use to generate the graph properties.
```

| -v (--verbose)
| Enables verbose mode for debugging purposes.
| false
|===

== Graph templates

The JMX Config Generator uses a template file to generate graphs.
It is possible to use a user-defined template.

> The `--template` option, followed by a file name, lets the JMX Config Generator use an external template file as base for graph generation.

Reviewer: ```suggestion
The `--template` option, followed by a file name, lets the JMX Config Generator use an external template file as the base for graph generation.
```

The following example shows how the custom `mytemplate.vm` file is used to generate the `mygraph.properties` graph definition file, using the configuration settings defined in `myconfig.xml`:

[source, console]
----
./jmx-config-generator generate-graph --input myconfig.xml --output mygraph.properties --template mytemplate.vm
----

> The template file must be an http://velocity.apache.org[_Apache Velocity_] template.

Reviewer: ```suggestion
The template file must be an http://velocity.apache.org[Apache Velocity] template.
```
